config/metric.go

```diff
@@ -0,0 +1,27 @@
+package config
+
+import (
+	"fmt"
+)
+
+const (
+	defaultMetricServerPort = 2112
+)
+
+// MetricConfig defines the server's basic configuration
+type MetricConfig struct {
+	MetricServerPort int `mapstructure:"metric-server-port"`
```

Comment:
```suggestion
	ServerPort int `mapstructure:"server-port"`
```

since the config struct already mentions metrics